arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxInitializationSection.java

```diff
@@ -142,80 +129,82 @@ public TxInitializationSection(Hub hub, WorldView world) {
-          .incrementBalanceBy(incrementToApplyToRecipientBalance)
+          .incrementBalanceBy(value)
           .code(initCode)
           .turnOnWarmth()
           .setDeploymentInfo(deploymentInfo);
     } else {
-      recipientValueReceptionNew
-          .incrementBalanceBy(incrementToApplyToRecipientBalance)
-          .turnOnWarmth();
+      recipientValueReceptionNew.incrementBalanceBy(value).turnOnWarmth();
     }
     recipientUndoingValueReception = recipientValueReceptionNew.deepCopy();
 
-    recipientDomSubStamps = DomSubStampsSubFragment.standardDomSubStamps(hub.stamp(), 1);
-
-    final TransactionFragment txFragment = TransactionFragment.prepare(tx);
-
-    // MISC i+0
-    this.addFragments(ImcFragment.forTxInit(hub));
+    this.addFragments(ImcFragment.forTxInit(hub)); // MISC i + 0
+    this.addFragment(TransactionFragment.prepare(tx)); // TXN i + 1
 
-    // TXN i+1
-    this.addFragment(txFragment);
+    this.addFragment( // ACC i + 2 (sender: gas payment)
+        accountFragmentFactory.makeWithTrm(
+            senderGasPayment,
+            senderGasPaymentNew,
+            senderAddress,
+            DomSubStampsSubFragment.standardDomSubStamps(hubStamp, 0)));
 
-    accountFragmentFactory = hub.factories().accountFragment();
-
-    // ACC i+2 (sender)
-    this.addFragment(
-        accountFragmentFactory.make(senderGasPayment, senderGasPaymentNew, senderDomSubStamps));
-
-    // ACC i+3 (sender)
-    this.addFragment(
+    this.addFragment( // ACC i + 3 (sender: value transfer)
         accountFragmentFactory.make(
-            senderValueTransfer, senderValueTransferNew, senderDomSubStamps));
+            senderValueTransfer,
+            senderValueTransferNew,
+            DomSubStampsSubFragment.standardDomSubStamps(hubStamp, 1)));
 
-    // ACC i+4 (recipient)
-    this.addFragment(
+    this.addFragment( // ACC i + 4 (recipient: value reception)
         accountFragmentFactory
             .makeWithTrm(
                 recipientValueReception,
                 recipientValueReceptionNew,
-                // Note that this depends on weather the sender is equal to the recipient
                 recipientAddress,
-                recipientDomSubStamps)
+                DomSubStampsSubFragment.standardDomSubStamps(hub.stamp(), 2))
             .requiresRomlex(true));
 
+    initializationContextFragment = ContextFragment.initializeExecutionContext(hub);
+
     hub.state.setProcessingPhase(TX_EXEC);
   }
 
   @Override
   public void resolvePostTransaction(
       Hub hub, WorldView state, Transaction tx, boolean isSuccessful) {
+
     if (!isSuccessful) {
-      senderUndoingValueTransferNew =
-          senderUndoingValueTransfer.deepCopy().setDeploymentNumber(hub).incrementBalanceBy(value);
 
+      senderUndoingValueTransfer = senderValueTransferNew.deepCopy().setDeploymentNumber(hub);
+      senderUndoingValueTransferNew = senderValueTransfer.deepCopy().setDeploymentNumber(hub);
```

Comment:
Doing things (_de_crementing by `value`) in reverse order amounts to doing the opposite (_in_crementing by `value`)